pyedr/pyedr/pyedr.py

```diff
@@ -461,10 +469,29 @@ def read_edr(path: str, verbose: bool = False) -> read_edr_return_type:
               end='', file=sys.stderr)
         print('\n{} frame read in {:.2f} seconds'.format(ifr, end - begin),
               file=sys.stderr)
-    return all_energies, all_names, times
+    return all_energies, all_names, times, unit_dict
 
 
-def edr_to_dict(path: str, verbose: bool = False) -> Dict[str, np.ndarray]:
+def get_unit_dictionary(path: str) -> Dict[str, str]:
+    """Calls :func:`read_edr` and returns the dictionary containing unit
+    information, mapping column names (str) to unit names (str).
+
+    Parameters
+    ----------
+    path : str
+        path to EDR file to be read
+
+    Returns
+    -------
+    unit_dict: Dict[str, str]
+        A dictionary mapping the term names to their units.
+    """
+    _, _, _, unit_dict = read_edr(path)
```

Comment:
Do you have to read **the whole file** just to get the units? That seems excessive. 

If the units are just
```python
    unit_dict = {'Time': "ps"}
    for nm in edr_file.nms:
        unit_dict[nm.name] = nm.unit
```
then why not write a function to just get the units and then call this function as part of `read_edr()`?

Or cache... if you cache then I'd not use the `get_*` method name but just make `unit_dict` a managed attribute.